applications/CoSimulationApplication/python_scripts/solver_wrappers/kratos/kratos_base_wrapper.py

```diff
@@ -26,38 +26,39 @@ def __init__(self, settings, solver_name):
         with open(input_file_name,'r') as parameter_file:
             self.project_parameters = KM.Parameters(parameter_file.read())
 
+        self._analysis_stage = self._CreateAnalysisStage()
+        # TODO add historical variables (or check if they are already there? => could maybe be done in check)
+        # maybe print warning if they are not there yet ...
+        # Note that it should also be considered if it is coming from a coupling solver
+        self._AllocateHistoricalVariablesFromCouplingData()
+
     def Initialize(self):
-        self._GetAnalysisStage().Initialize()
+        self._analysis_stage.Initialize() # this reades the Meshes
         super(KratosBaseWrapper, self).Initialize()
 
     def Finalize(self):
-        self._GetAnalysisStage().Finalize()
+        self._analysis_stage.Finalize()
 
     def AdvanceInTime(self, current_time):
-        new_time = self._GetAnalysisStage()._GetSolver().AdvanceInTime(current_time)
-        self._GetAnalysisStage().time = new_time # only needed to print the time correctly
+        new_time = self._analysis_stage._GetSolver().AdvanceInTime(current_time)
+        self._analysis_stage.time = new_time # only needed to print the time correctly
         return new_time
 
     def InitializeSolutionStep(self):
-        self._GetAnalysisStage().InitializeSolutionStep()
+        self._analysis_stage.InitializeSolutionStep()
 
     def Predict(self):
-        self._GetAnalysisStage()._GetSolver().Predict()
+        self._analysis_stage._GetSolver().Predict()
 
     def SolveSolutionStep(self):
-        self._GetAnalysisStage()._GetSolver().SolveSolutionStep()
+        self._analysis_stage._GetSolver().SolveSolutionStep()
 
     def FinalizeSolutionStep(self):
-        self._GetAnalysisStage().FinalizeSolutionStep()
+        self._analysis_stage.FinalizeSolutionStep()
```

Comment:
why did you remove the function `_GetAnalysisStage` ? Isn't more safe ?

Reply:
this is related to the allocation of the historical variables
=> to be discussed in person